x-pack/filebeat/input/httpjson/input.go

```diff
@@ -81,532 +48,145 @@ func newRetryLogger() *retryLogger {
-				mm, err = in.processEventArray([]interface{}{obj})
-				if err != nil {
-					return err
-				}
-			} else {
-				v, err = common.MapStr(obj).GetValue(in.config.JSONObjects)
-				if err != nil {
-					if err == common.ErrKeyNotFound {
-						break
-					}
-					return err
-				}
-				switch ts := v.(type) {
-				case []interface{}:
-					mm, err = in.processEventArray(ts)
-					if err != nil {
-						return err
-					}
-				default:
-					return errors.Errorf("content of %s is not a valid array", in.config.JSONObjects)
-				}
-			}
-		default:
-			in.log.Debug("http.response.body is not a valid JSON object", string(responseData))
-			return errors.Errorf("http.response.body is not a valid JSON object, but a %T", obj)
+		switch in.config.URL.Scheme {
+		case "https":
+			return "443"
 		}
+		return "80"
+	}()
 
-		if mm != nil && in.config.Pagination.IsEnabled() {
-			if in.config.Pagination.Header != nil {
-				// Pagination control using HTTP Header
-				url, err := getNextLinkFromHeader(header, in.config.Pagination.Header.FieldName, in.config.Pagination.Header.RegexPattern)
-				if err != nil {
-					return errors.Wrapf(err, "failed to retrieve the next URL for pagination")
-				}
-				if ri.URL == url || url == "" {
-					in.log.Info("Pagination finished.")
-					break
-				}
-				ri.URL = url
-				if err = in.applyRateLimit(ctx, header, in.config.RateLimit); err != nil {
-					return err
-				}
-				in.log.Info("Continuing with pagination to URL: ", ri.URL)
-				continue
-			} else {
-				// Pagination control using HTTP Body fields
-				ri, err = createRequestInfoFromBody(in.config.Pagination, common.MapStr(response), common.MapStr(mm), ri)
-				if err != nil {
-					return err
-				}
-				if ri == nil {
-					break
-				}
-				if err = in.applyRateLimit(ctx, header, in.config.RateLimit); err != nil {
-					return err
-				}
-				in.log.Info("Continuing with pagination to URL: ", ri.URL)
-				continue
-			}
-		}
-		break
-	}
-
-	if mm != nil && in.config.DateCursor.IsEnabled() {
-		in.advanceCursor(common.MapStr(mm))
+	_, err := net.DialTimeout("tcp", net.JoinHostPort(in.config.URL.Hostname(), port), time.Second)
+	if err != nil {
+		return fmt.Errorf("url %q is unreachable", in.config.URL)
 	}
 
 	return nil
 }
 
-func (in *HttpjsonInput) getURL() string {
-	if !in.config.DateCursor.IsEnabled() {
-		return in.config.URL
-	}
+// Run starts the input worker then returns. Only the first invocation
+// will ever start the worker.
+func (in *httpJSONInput) Run(ctx v2.Context, publisher stateless.Publisher) error {
+	log := ctx.Logger.With("url", in.config.URL)
 
-	var dateStr string
-	if in.nextCursorValue == "" {
-		t := timeNow().UTC().Add(-in.config.DateCursor.InitialInterval)
-		dateStr = t.Format(in.config.DateCursor.GetDateFormat())
-	} else {
-		dateStr = in.nextCursorValue
-	}
+	stdCtx := ctxtool.FromCanceller(ctx.Cancelation)
 
-	url, err := url.Parse(in.config.URL)
+	httpClient, err := in.newHTTPClient(stdCtx)
```

Comment:
how can this fail? Does the client make a request to init oauth support? Are there recoverable error types (e.g. io timeout)? In the later case we might want to put the initialization into a loop with exponential backoff.